core/src/main/scala/org/apache/spark/sql/crossdata/catalog/CatalogChain.scala

```diff
@@ -125,16 +125,30 @@ private[crossdata] class CatalogChain private(val temporaryCatalogs: Seq[XDTempo
   override def persistView(tableIdentifier: ViewIdentifier, plan: LogicalPlan, sqlText: String): Unit =
     persistentCatalogs.foreach(_.saveView(tableIdentifier, plan, sqlText))
 
+  override def persistIndex(crossdataIndex: CrossdataIndex): Unit =
+    if (persistentChainedLookup(_.lookupTable(crossdataIndex.tableIdentifier)).isEmpty) {
```

Comment:
You could call `tableMetadata` here so you can get updates of the table look-up at persistent catalog behavior